x/data/client/tx.go

```diff
@@ -20,27 +26,141 @@ func TxCmd() *cobra.Command {
 
 	cmd.AddCommand(
 		MsgAnchorDataCmd(),
+		MsgSignDataCmd(),
+		MsgStoreDataCmd(),
 	)
 
 	return cmd
 }
 
+// MsgAnchorDataCmd creates a CLI command for Msg/AnchorData.
 func MsgAnchorDataCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use: "send [from_key_or_address] [to_address] [amount]",
-		Short: `Send funds from one account to another. Note, the'--from' flag is
-ignored as it is implied from [from_key_or_address].`,
-		Args: cobra.ExactArgs(3),
+		Use: "anchor [sender] [cid]",
+		Short: "Anchors a piece of data to the blockchain based on its secure " +
+			"hash, effectively providing a tamper resistant timestamp.",
+		Args: cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := cmd.Flags().Set(flags.FlagFrom, args[0])
+			if err != nil {
+				return err
+			}
+
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err = client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			cid, err := gocid.Decode(args[1])
+			if err != nil {
+				return err
+			}
+
+			msg := data.MsgAnchorDataRequest{
+				Sender: clientCtx.GetFromAddress().String(),
+				Cid:    cid.Bytes(),
+			}
+			svcMsgClientConn := &util.ServiceMsgClientConn{}
+			msgClient := data.NewMsgClient(svcMsgClientConn)
+			_, err = msgClient.AnchorData(context.Background(), &msg)
+			if err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), svcMsgClientConn.Msgs...)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// MsgSignDataCmd creates a CLI command for Msg/SignData.
+func MsgSignDataCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "sign [signer] [cid]",
+		Short: `Sign an arbitrary piece of data on the blockchain.`,
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := cmd.Flags().Set(flags.FlagFrom, args[0])
+			if err != nil {
+				return err
+			}
+
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err = client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			cid, err := gocid.Decode(args[1])
+			if err != nil {
+				return err
+			}
+
+			msg := data.MsgSignDataRequest{
+				Signers: []string{clientCtx.GetFromAddress().String()},
+				Cid:     cid.Bytes(),
+			}
+			svcMsgClientConn := &util.ServiceMsgClientConn{}
+			msgClient := data.NewMsgClient(svcMsgClientConn)
+			_, err = msgClient.SignData(context.Background(), &msg)
```

Comment:
We shouldn't use `context.Background`. Let's use `cmd.Context()`

Reply:
ack